gate-api/src/main/java/com/netflix/spinnaker/gate/api/extension/HttpRequest.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.api.extension;
+
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import java.util.Map;
+import lombok.Data;
+
+/** An {@code HttpRequest} represents a request that will be handled by an {@code ApiExtension}. */
+@Alpha
+@Data(staticConstructor = "of")
+public class HttpRequest {
```

Comment:
This is an incomplete contract right now ... is there an opportunity to use https://github.com/spinnaker/kork/blob/master/kork-plugins-api/src/main/java/com/netflix/spinnaker/kork/plugins/api/httpclient/Request.java?

Originally I was hoping to use standard `HttpServletRequest` and `HttpServletResponse` but wasn't sure how that would play with the shadowed equivalents in Spring Boot.

Reply:
Should we share Request and Response objects between client (plugin) and server?  I guess I'm not sure.

If we were to share them, I think a pretty simple Kork refactor like adding the package `com.netflix.spinnaker.plugins.api.http` and moving the `Request` and `Response` objects there would suffice.

Reply:
Tbh I'm not entirely sure that we should -- but I didn't realize those classes existed when I threw `HttpRequest` and `HttpResponse` together.

Reply:
I'm leaning on no sharing, but since this is an alpha contract (even beta), we'd be able to shift things around if we decide to share.